File: Source/Alternet.UI.Pal/ValidatorText.cpp
#include "ValidatorText.h"

namespace Alternet::UI
{
	wxTextValidator* ValidatorText::ToTextValidator(void* handle)
	{
		return (wxTextValidator*) handle;
	}

	void ValidatorText::DeleteValidatorText(void* handle)
	{
		delete ToTextValidator(handle);
	}

	void* ValidatorText::CreateValidatorText(int64_t style)
	{
		return new wxTextValidator((long)style);
	}

	ValidatorText::ValidatorText() 
	{

	}
	
	ValidatorText::~ValidatorText()
	{
	}

	string ValidatorText::IsValid(void* handle, const string& val)
	{
		return wxStr(ToTextValidator(handle)->IsValid(wxStr(val)));
	}

	int64_t ValidatorText::GetStyle(void* handle)
	{
		return ToTextValidator(handle)->GetStyle();
	}

	void ValidatorText::SetStyle(void* handle, int64_t style)
	{
		ToTextValidator(handle)->SetStyle(style);
	}

	void ValidatorText::SetCharIncludes(void* handle, const string& chars)
	{
		ToTextValidator(handle)->SetCharIncludes(wxStr(chars));
	}

	void ValidatorText::AddCharIncludes(void* handle, const string& chars)
	{
		ToTextValidator(handle)->AddCharIncludes(wxStr(chars));
	}

	string ValidatorText::GetCharIncludes(void* handle)
	{
		return wxStr(ToTextValidator(handle)->GetCharIncludes());
	}

	void ValidatorText::AddInclude(void* handle, const string& include)
	{
		ToTextValidator(handle)->AddInclude(wxStr(include));
	}

	void ValidatorText::AddExclude(void* handle, const string& exclude)
	{
		ToTextValidator(handle)->AddExclude(wxStr(exclude));
	}

	void ValidatorText::SetCharExcludes(void* handle, const string& chars)
	{
		ToTextValidator(handle)->SetCharExcludes(wxStr(chars));
	}

	void ValidatorText::AddCharExcludes(void* handle, const string& chars)
	{
		ToTextValidator(handle)->AddCharExcludes(wxStr(chars));
	}

	string ValidatorText::GetCharExcludes(void* handle)
	{												   
		return wxStr(ToTextValidator(handle)->GetCharExcludes());
	}

	void ValidatorText::ClearExcludes(void* handle)
	{
		wxArrayString a = wxArrayString();
		ToTextValidator(handle)->SetExcludes(a);
	}

	void ValidatorText::ClearIncludes(void* handle)
	{
		wxArrayString a = wxArrayString();
		ToTextValidator(handle)->SetIncludes(a);
	}
}

File: Source/NativeApi/NativeApi/Api/Validator/ValidatorText.cs
#pragma warning disable
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NativeApi.Api
{
    //https://docs.wxwidgets.org/3.2/classwx_text_validator.html
    public class ValidatorText : Validator
    {
        public static void DeleteValidatorText(IntPtr handle) { }
        public static IntPtr CreateValidatorText(long style) => default;

        public static long GetStyle(IntPtr handle) => default;
        public static void SetStyle(IntPtr handle, long style) { }

        public static void SetCharIncludes(IntPtr handle, string chars) { }
        public static void AddCharIncludes(IntPtr handle, string chars) { }
        public static string GetCharIncludes(IntPtr handle) => default;

        public static void AddInclude(IntPtr handle, string include) { }
        public static void AddExclude(IntPtr handle, string exclude) { }

        public static void SetCharExcludes(IntPtr handle, string chars) { }
        public static void AddCharExcludes(IntPtr handle, string chars) { }
        public static string GetCharExcludes(IntPtr handle) => default;

        public static void ClearExcludes(IntPtr handle) { }
        public static void ClearIncludes(IntPtr handle) { }

        // Returns the error message if the contents of val are invalid or
        // the empty string if val is valid.
        public static string IsValid(IntPtr handle, string val) => default;
    }
}